File: benchmarks/bench_lsearch.zig
fn lsearch(a: []const i32, val: i32) i32 {
    for (a, 0..) |x, i| if (x == val) return @intCast(i);
    return -1;
}
pub fn main() u8 {
    const a = [10]i32{3,7,1,9,5,2,8,4,6,0};
    var r: i32 = 0;
    var i: usize = 0;
    while (i < 10000000) : (i += 1) r += lsearch(&a, 5);
    return if (r == 40000000) 0 else 1;
}

File: benchmarks/bench_lsearch.c
#include <stdint.h>
static int lsearch(const int32_t *a, int n, int val) {
    for (int i=0; i<n; i++) if (a[i]==val) return i;
    return -1;
}
int main(void) {
    int32_t a[10]={3,7,1,9,5,2,8,4,6,0};
    int r = 0;
    for (int i=0; i<10000000; i++) r += lsearch(a,10,5);
    return r == 40000000 ? 0 : 1;
}
